frontend/src/app/BibWindow.js

```diff
@@ -162,8 +165,8 @@ class BibWindow extends React.Component {
         if (source.title) {
             title = title + source.title + ".";
         }
-        if (source.author) {
-            author = author + source.author + ".";
+        if (source.lastName && source.firstName) {
+            author = author + source.lastName + ", " + source.firstName + ".";
```

Comment:
Small thing, but since these lines are repeated 3 times, it would be ideal to extract them to a helper function inside this class. I think the final call would be `author = this.formatAuthor(source, author)`, and the function either returns the same input author if first or last name are empty, or returns the constructed field if they're not empty. It seems like there are other parts of this files that have a similar issue, it'd be cool if you could maybe fix that.

Reply:
Ask and ye shall receive